pkg/storage/mysql/mysqlstore.go

```diff
@@ -0,0 +1,271 @@
+	}
+
+	p.dbs = make(map[string]*sqlDBStore)
+
+	return nil
+}
+
+// CloseStore closes a previously opened store
+func (p *Provider) CloseStore(name string) error {
+	p.Lock()
+	defer p.Unlock()
+
+	store, exists := p.dbs[name]
+	if !exists {
+		return nil
+	}
+
+	delete(p.dbs, name)
+
+	conn, err := store.db.Conn(context.Background())
+	if err != nil {
+		return err
+	}
+
+	return conn.Close()
+}
+
+// Put stores the key and the record
+func (s *sqlDBStore) Put(k string, v []byte) error {
+	if k == "" || v == nil {
+		return errors.New("key and value are mandatory")
+	}
+
+	createTableStmt := "CREATE Table IF NOT EXISTS `tablePrefix s.dbName` (`key` varchar(50) NOT NULL, `value` BLOB, " +
+		"PRIMARY KEY (`key`));"
+
+	// creating key-value table inside the database
+	stmt, err := s.db.Prepare(createTableStmt)
+
+	if err != nil {
+		return fmt.Errorf("failed to create table: %w", err)
+	}
+
+	// executing the prepared create table statement
+	_, err = stmt.Exec()
+	if err != nil {
+		return fmt.Errorf("failed to execute the create table statement: %w", err)
+	}
+
+	// prepare upsert query to insert the record, checking whether the key is already mapped to a value in the store.
+	queryStmt, err := s.db.Prepare("INSERT INTO `tablePrefix s.dbName` VALUES (?, ?) ON DUPLICATE KEY " +
+		"UPDATE value=? ")
+	if err != nil {
+		return fmt.Errorf("failed to insert the record in the table %s: %w", `tablePrefix s.dbName`, err)
+	}
+
+	// executing the prepared upsert statement
+	_, err = queryStmt.Exec(k, v, v)
+	if err != nil {
+		return fmt.Errorf("failed to execute the insert query statement %w", err)
+	}
+
+	return queryStmt.Close()
+}
+
+// Get fetches the record based on key
+func (s *sqlDBStore) Get(k string) ([]byte, error) {
+	if k == "" {
+		return nil, errors.New("key is mandatory")
+	}
+
+	var res result
+
+	err := s.db.QueryRow("SELECT * FROM `tablePrefix s.dbName` WHERE `key`=?", k).Scan(&res.key, &res.value)
+	if err != nil {
+		if strings.Contains(err.Error(), sqlDBNotFound) {
+			return nil, storage.ErrDataNotFound
+		}
+
+		return nil, err
+	}
+
+	return res.value, nil
+}
+
+// Delete will delete record with k key
+func (s *sqlDBStore) Delete(k string) error {
+	if k == "" {
+		return errors.New("key is mandatory")
+	}
+
+	res, err := s.db.Exec("DELETE FROM `tablePrefix s.dbName` WHERE `key`=?", k)
+	if err == nil {
+		count, e := res.RowsAffected()
+		if e == nil {
+			if count == 0 {
+				return errors.New("failed to delete row")
+			}
+
+			return nil
+		}
+	}
+
+	return err
+}
+
+type sqlDBResultsIterator struct {
+	store      *sqlDBStore
+	resultRows *sql.Rows
+	err        error
+}
+
+func (s *sqlDBStore) Iterator(startKey, endKey string) storage.StoreIterator {
+	resultRows, err := s.db.Query("SELECT * FROM `tablePrefix s.dbName` WHERE `key` LIKE ?", startKey+"%")
+	if err != nil {
+		if err = resultRows.Err(); err != nil {
+			return &sqlDBResultsIterator{store: s, resultRows: &sql.Rows{},
+				err: fmt.Errorf("no rows found")}
```

Comment:
please confirm if this is behavour is expected

Reply:
`endKey ` here is unused. Replace `endKey` with`_`. The query will not work. 
As I know you need to find data between `startKey and endKey`. `LIKE` doesn't do that.

Reply:
@talwinder50  The simplest idea is to use `BETWEEN` operator. You need to create the following table:

|  ID   | KEY | VAL |
| ---- | ---- | ---- |
| 1      | start | v1    |
| 2      | end | v1    |
Where :
ID - should be an integer (auto-increment).
KEY - unique index 

```
SELECT * FROM table
WHERE key BETWEEN "start" AND "end"
ORDER BY ID;
```

Notice that we also need to find data by the key which is not fully presented. e.g `start` -> `sta`.
In that case, you can extend the query above with the subquery (using `LIKE` operator).

Reply:
@soluchok why should we order by `ID` instead of `key`?

Reply:
@llorllale As I know we need to keep consistency. My point is to add an ID(auto increment) which will guarantee the sequence.
But probably it will even work without an ID field. I did not check it.

Reply:
It will work without the `ID` field. The `k` arg is the key'.

Reply:
Have updated this using sub query to simulate the range. Unit test is updated